tests/tests_geomstats/test_special_euclidean.py

```diff
@@ -1,330 +1,1230 @@
+            return self._geodesic_bvp_belongs_test_data(
+                self.metric_args_list,
+                self.space_list,
+                self.n_points_list,
+                belongs_atol=gs.atol * 100,
+            )
+
+        def log_exp_composition_test_data(self):
+            return self._log_exp_composition_test_data(
+                self.metric_args_list,
+                self.space_list,
+                self.n_samples_list,
+                rtol=gs.rtol * 100,
+                atol=gs.atol * 100,
+            )
+
+        def exp_log_composition_test_data(self):
+            return self._exp_log_composition_test_data(
+                self.metric_args_list,
+                self.space_list,
+                self.shape_list,
+                self.n_samples_list,
+                amplitude=10,
+                rtol=gs.rtol * 100,
+                atol=gs.atol * 100,
+            )
+
+        def exp_ladder_parallel_transport_test_data(self):
+            return self._exp_ladder_parallel_transport_test_data(
+                self.metric_args_list,
+                self.space_list,
+                self.shape_list,
+                self.n_samples_list,
+                self.n_rungs_list,
+                self.alpha_list,
+                self.scheme_list,
+            )
+
+        def exp_geodesic_ivp_test_data(self):
+            return self._exp_geodesic_ivp_test_data(
+                self.metric_args_list,
+                self.space_list,
+                self.shape_list,
+                self.n_samples_list,
+                self.n_points_list,
+                rtol=gs.rtol * 100,
+                atol=gs.atol * 100,
+            )
+
+        def parallel_transport_ivp_is_isometry_test_data(self):
+            return self._parallel_transport_ivp_is_isometry_test_data(
+                self.metric_args_list,
+                self.space_list,
+                self.shape_list,
+                self.n_samples_list,
+                is_tangent_atol=gs.atol * 1000,
+                atol=gs.atol * 1000,
+            )
+
+        def parallel_transport_bvp_is_isometry_test_data(self):
+            return self._parallel_transport_bvp_is_isometry_test_data(
+                self.metric_args_list,
+                self.space_list,
+                self.shape_list,
+                self.n_samples_list,
+                is_tangent_atol=gs.atol * 1000,
+                atol=gs.atol * 1000,
+            )
+
+    testing_data = SpecialEuclideanMatrixCanonicalLeftMetricTestData()
+
+    def test_left_metric_wrong_group(self, group, expected):
+        with expected:
+            self.metric(group)
 
-    def test_right_exp_coincides(self):
-        vector_group = SpecialEuclidean(n=2, point_type="vector")
-        theta = gs.pi / 2
-        initial_vec = gs.array([theta, 1.0, 1.0])
-        initial_matrix_vec = self.group.lie_algebra.matrix_representation(initial_vec)
+
+class TestInvariantMetricOnSE(
```

Comment:
This should be: TestSpecialEuclideanMatrixCanonicalRightMetric to be consistent.

Also, correct typo in this file: Canonical (with one 'n').